iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/NetworkXml.java

```diff
@@ -157,18 +173,31 @@ private static void writeExtension(Extension<? extends Identifiable<?>> extensio
         if (extensionXmlSerializer == null) {
             throw new AssertionError("Extension XML Serializer of " + extension.getName() + " should not be null");
         }
-
-        String namespaceUri = getNamespaceUri(extensionXmlSerializer, context.getOptions(), context.getVersion());
-
-        if (extensionXmlSerializer.hasSubElements()) {
-            writer.writeStartElement(namespaceUri, extension.getName());
-        } else {
-            writer.writeEmptyElement(namespaceUri, extension.getName());
+        boolean versionExist = true;
+        if (extensionXmlSerializer instanceof AbstractVersionableNetworkExtensionXmlSerializer) {
```

Comment:
It's a little beyond the scope of that issue, but I think instead of the abstract class we should have an sub-interface for XIIDM extensions serializers.

The `ExtensionXmlSerializer` interface is too generic for the XIIDM export, because it does not provide the mapping between XIIDM version and extension version, which is necessary.

It would simplify the code in here.

Reply:
Actually there is a quite complex logic in that `writeExtension` method, in particular for version handling, that we apply to each of the extension object we are serializing:
it would be better to execute it only once for each extension class we want to serialize!
For example by applying a preprocessing to each serializer to instantiate an object where version is well defined, and re-use it afterwards.